packages/block-library/src/navigation/edit/placeholder/index.js

```diff
@@ -45,7 +47,39 @@ export default function NavigationPlaceholder( {
 		hasMenus,
 	} = useNavigationEntities();
 
-	const isStillLoading = isResolvingPages || isResolvingMenus;
+	const {
+		dispatch: convertClassicMenuToBlocks,
+		blocks: classicMenuBlocks,
+		name: classicMenuName,
+		isResolving: isResolvingClassicMenuConversion,
+		hasResolved: hasResolvedClassicMenuConversion,
+	} = useConvertClassicMenu();
+
+	useEffect( () => {
+		async function handleCreateNav() {
+			const navigationMenuPost = await createNavigationMenu(
+				classicMenuName,
+				classicMenuBlocks
+			);
+			onFinish( navigationMenuPost, classicMenuBlocks );
+		}
+		if (
+			hasResolvedClassicMenuConversion &&
+			classicMenuName &&
+			classicMenuBlocks?.length
+		) {
+			handleCreateNav();
+		}
+	}, [
+		classicMenuBlocks,
+		classicMenuName,
+		hasResolvedClassicMenuConversion,
+	] );
+
+	const isStillLoading =
```

Comment:
I know this was already in `trunk`, but I think a good improvement would be a rename of this variable:
```suggestion
	const isLoading =
```